packages/database/src/migrations/20210623032402-FixStockCardMatrixChartTonga-modifies-data.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const dashboardId = 'UNFPA_RH_Stock_Cards_Use_Matrix';
+
+exports.up = async function (db) {
+  await db.runSql(`
+      update "dashboardReport" dr
+      set "viewJson" = regexp_replace(dr."viewJson"::text, '\\"presentationOptions\\"\\: \\{','"presentationOptions" : {"type":"condition",','g')::jsonb
+      where id = '${dashboardId}';
+
+      update "dashboardReport" dr
+      set "viewJson" = regexp_replace(dr."viewJson"::text, '\\"key\\"\\: 0','"condition": "No"','g')::jsonb
```

Comment:
I'm not sure if Yes/No and 1/0 is the problem here... We've already sanitized internal data to use 1/0. So the current config should have worked. I'm suspecting it's not working because of something else

Reply:
As discussed, even though raw values are returned as 1/0, `tableOfDataValues` always translate the value using metadata options, plus Yes/No is always returned as default binary options for internal data. So that's why `tableOfDataValues` will always translate 1/0 to Yes/No. 

I suspect there are more matrices using binary data that are having the same problem, and we just haven't found out yet. So a generic solution would be preferred instead of fixing by changing the config in multiple visualisations. We have a couple of solutions:

- Investigate why we always need to translate values to the metadata options in `tableOfDataValues` (we don't do that in any other data builders). I think this is related how we build matrices using DHIS2 data. We've converted a lot of DHIS2 data to internal. So if it's no longer relevant, we can remove it entirely?

- Check in `tableOfDataValues` if the metadata options is the default Yes/No, if it is then don't perform the options translation.

Reply:
Thanks @chris-bes ! I am doing the investigation, and one concern is that if we want to download the matrix it will show 1/0 in spreadsheet, but that is not a major issue for now.